src/server/game/AI/ScriptedAI/ScriptedCreature.h

```diff
@@ -197,6 +197,66 @@ struct ScriptedAI : public CreatureAI
     // Called when AI is temporarily replaced or put back when possess is applied or removed
     void OnPossess(bool /*apply*/) {}
 
+    enum class Axis
+    {
+        AXIS_X,
+        AXIS_Y
+    };
+
+    /* This is called for bosses whenever an encounter is happening.
+     * - Arguments:
+     * - Position has to be passed as a constant pointer (&Position)
+     * - Axis is the X or Y axis that is used to decide the position threshold
+     * - Above decides if the boss position should be above the passed position
+     *   or below.
+     * Example:
+     * Hodir is in room until his Y position is below the Door position:
+     * IsInRoom(doorPosition, AXIS_Y, false);
+     */
+    bool IsInRoom(const Position* pos, Axis axis, bool above)
+    {
+        if (!pos)
+        {
+            return true;
+        }
+
+        switch (axis)
+        {
+            case Axis::AXIS_X:
+                if (!above)
+                {
+                    if (me->GetPositionX() < pos->GetPositionX())
+                    {
+                        EnterEvadeMode();
+                        return false;
+                    }
+                }
```

Comment:
```suggestion
                if (!above && me->GetPositionX() < pos->GetPositionX())
                {
                    EnterEvadeMode();
                    return false;
                }
```

Reply:
add in the condition `|| me->GetPositionX() > pos->GetPositionX()` and remove the code below